packages/neon_framework/packages/account_repository/lib/src/account_storage.dart

```diff
@@ -3,52 +3,205 @@ part of 'account_repository.dart';
 /// {@template account_repository_storage}
 /// Storage for the [AccountRepository].
 /// {@endtemplate}
-@immutable
-class AccountStorage {
+class AccountStorage with Table {
   /// {@macro account_repository_storage}
-  const AccountStorage({
-    required this.accountsPersistence,
-    required this.lastAccountPersistence,
-  });
+  AccountStorage();
 
-  /// The store for the account list.
-  final SingleValueStore accountsPersistence;
+  @override
+  String get name => 'account_credentials';
 
-  /// The store for the last used account.
-  final SingleValueStore lastAccountPersistence;
+  @override
+  int get version => 1;
+
+  /// Column for the [Credentials.serverURL].
+  @protected
+  static const String serverURL = 'server_urlL';
+
+  /// Column for the [Credentials.username].
+  @protected
+  static const String loginName = 'login_name';
+
+  /// Column for the [Credentials.appPassword].
+  @protected
+  static const String appPassword = 'app_password';
+
+  /// Column to store the active account.
+  @protected
+  static const String active = 'active';
+
+  @override
+  void onCreate(Batch db, int version) {
+    db.execute('''
+CREATE TABLE "$name" (
+  "$serverURL"      TEXT NOT NULL,
+  "$loginName"      TEXT NOT NULL,
+  "$appPassword"    TEXT,
+  "$active"         BOOLEAN NOT NULL DEFAULT 0,
+  PRIMARY KEY("$serverURL", "$loginName")
+);
+''');
+  }
+
+  static final Logger _log = Logger('AccountStorage');
+
+  Database get _database => controller.database;
 
   /// Gets a list of logged in credentials from storage.
   ///
   /// It is not checked whether the stored information is still valid.
   Future<BuiltList<Credentials>> readCredentials() async {
-    if (accountsPersistence.hasValue()) {
-      return accountsPersistence
-          .getStringList()!
-          .map((a) => Credentials.fromJson(json.decode(a) as Map<String, dynamic>))
-          .toBuiltList();
+    try {
+      final results = await _database.query(
+        name,
+        columns: [
+          serverURL,
+          loginName,
+          appPassword,
+        ],
+      );
+
+      final credentials = ListBuilder<Credentials>();
+
+      for (final result in results) {
+        credentials.add(
+          Credentials((b) {
+            b
+              ..serverURL = Uri.parse(result[serverURL]! as String)
+              ..username = result[loginName]! as String
+              ..appPassword = result[appPassword] as String?;
+          }),
+        );
+      }
+
+      return credentials.build();
+    } on DatabaseException catch (error, stackTrace) {
+      _log.warning(
+        'Error loading cookies.',
```

Comment:
Wrong error message (same for all other catch blocks in this class)

Reply:
The neon_storage package handles this with a conditional import/export.

https://github.com/nextcloud/neon/blob/54840498a0a1e5c96721f878a4a4a653831e5bfe/packages/neon_framework/packages/neon_storage/lib/src/sqlite/sqlite.dart#L1-L3